### PR TITLE
fix: Error reporting for artifacts.

### LINKED ISSUE #518: Strange doubled message when listing artifacts without id passed
**Describe the bug**

➜  testkube git:(main) t artifacts list
Error: execution-id is a required parameter
execution-id is a required parameter

Use common ui framework for messages and infos

### PR DESCRIPTION
**Pull request by jasmingacic**
Strange doubled message when listing artifacts without id passed #518

Signed-off-by: jasmingacic <jasmin.gacic@gmail.com>

### CODE REVIEW

**Comment by exu** [comment]
Please add PR title (ideally with conventional commit fix: prefix )

File: internal/app/api/v1/scripts.go [hunk 1]
```diff
@@ -443,7 +444,9 @@ func (s testkubeAPI) ListArtifacts() fiber.Handler {
 	return func(c *fiber.Ctx) error {
 
 		executionID := c.Params("executionID")
-
+		if s.Storage == nil || reflect.ValueOf(s.Storage).IsNil() {
```
Comment (exu) [inline internal/app/api/v1/scripts.go:447]: Reflection looks odd

Comment (nicufk) [reply internal/app/api/v1/scripts.go:447]: Agree. Can we find a more elegant way?

Comment (jasmingacic) [reply internal/app/api/v1/scripts.go:447]: it is either of these options
https://play.golang.org/p/QlObAWkiVdB
i could cast s.Storage.(*minio.Client) == nil

File: pkg/api/v1/client/proxy.go [hunk 2]
```diff
@@ -443,12 +443,12 @@ func (c ProxyScriptsAPI) DownloadFile(executionID, fileName, destination string)
 		Suffix(uri).
 		SetHeader("Accept", "text/event-stream").
 		Stream(context.Background())
```
Comment (jasmingacic) [inline pkg/api/v1/client/proxy.go:445]: @exu how do you check for status code in such calls?

File: internal/app/api/v1/scripts.go (hunk 1 above)
Comment (jasmingacic) [reply internal/app/api/v1/scripts.go]: addressed it

File: pkg/api/v1/client/proxy.go (hunk 2 above)
Comment (nicufk) [reply pkg/api/v1/client/proxy.go:445]: @jasmingacic look into makeDeleteRequest, something like that
`		var code int
		resp.StatusCode(&code)
		if code != http.StatusNoContent {
			respBody, err := resp.Raw()
			if err != nil {
				return err
			}
			return fmt.Errorf("request returned error: %s", respBody)
		}`

**Review by exu — APPROVED** [review]
LGTM

File: pkg/api/v1/client/proxy.go (hunk 2 above)
Comment (jasmingacic) [reply pkg/api/v1/client/proxy.go:445]: I would if i had any response here.
Let's merge it and then we deal with it later

**Pull request merged**